integration-tests/bats/filter-branch.bats

```diff
@@ -427,3 +427,122 @@ SQL
     [[ "$output" =~ "| 3  | 3  | NULL   |" ]] || false
     [[ "$output" =~ "+----+----+--------+" ]] || false
 }
+
+@test "filter-branch: works with working and staged changes in other branch" {
+    dolt sql << SQL
+call dolt_checkout('-b', 'other');
+insert into test values (3, 3);
+SQL
+    run dolt filter-branch -q "alter table test add column filter int"
+    [ "$status" -eq 0 ]
+
+    run dolt sql -q "select * from test as of 'HEAD'"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "+----+----+--------+" ]] || false
```

Comment:
This format is pretty fragile, you should use the `-r csv` option to check row output from `dolt sql`

Reply:
Same comment in other places